components/common-go/baseserver/server.go

```diff
@@ -364,7 +364,7 @@ func newBuiltinServices(server *Server) *builtinServices {
 			Handler: server.healthEndpoint(),
 		},
 		Metrics: &http.Server{
-			Addr:    fmt.Sprintf(":%d", BuiltinMetricsPort),
+			Addr:    fmt.Sprintf("127.0.0.1:%d", BuiltinMetricsPort),
```

Comment:
This was the reason of the collision

Reply:
I've changed it away from localhost:port as that listened only on local host and couldn't be reached from outside the container. This should work also. Will check preview shortly.